### PR TITLE
Fixes for all-runtimes

### PR DESCRIPTION
**Pull request by RobertSamoilescu**
This PR includes some minor fixes for the `all-runtimes` tests.

### CODE REVIEW

File: mlserver/env.py [hunk 1]
```diff
@@ -117,6 +117,9 @@ def _lib_path(self) -> str:
         matches = glob.glob(pattern)
 
         for match in matches:
+            if match.endswith("3.1"):
```
Comment (sakoush) [inline mlserver/env.py:120]: why is this required? add a note in the code.

File: mlserver/metrics/server.py [hunk 2]
```diff
@@ -72,5 +73,8 @@ def _get_config(self):
         return uvicorn.Config(self._app, **kwargs)
 
     async def stop(self, sig: Optional[int] = None):
+        if sig is None:
```
Comment (sakoush) [inline mlserver/metrics/server.py:76]: why is this required? add a note in the code

File: tox.ini [hunk 3]
```diff
@@ -42,6 +42,10 @@ commands =
         {toxinidir}/runtimes/lightgbm \
         {toxinidir}/runtimes/mlflow \
         {toxinidir}/runtimes/huggingface
+set_env =
```
Comment (sakoush) [inline tox.ini:45]: ditto

**Review by sakoush — APPROVED** [review]
lgtm. please clarify in the code why these changes are added.

File: tox.ini (hunk 3 above)
Comment (RobertSamoilescu) [reply tox.ini:45]: See the `mlserver` env which requires that for some env test. Feels a bit weird to add some comments there.

Comment (sakoush) [reply tox.ini:45]: oh I see, we missed this change from [here](https://github.com/SeldonIO/MLServer/blob/b848836872a215cc781075afdbcac345bea49ab6/tox.ini#L24) to be added to `all-runtimes`

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -167,13 +167,11 @@ jobs:
 
   # Ensure that having all the runtimes installed together works
   all-runtimes:
-    if: github.event_name == 'push'
```
Comment (sakoush) [inline .github/workflows/tests.yml:170]: assuming this is for testing, revert back before merging.
